### DISCUSSION TITLE
Console Redirect into Avalonia Window

### DISCUSSION THREAD

**Discussion by sherlockson** [AvaloniaUI/Avalonia#6366] [Q&A] [answered]
Hi all,

I am currently converting a WPF application into Avalonia so that I can take advantage of the cross-platform capabilities of the framework. It's going well so far, but I have a question as to how I can implement a redirect for console output into the UI. Currently I have a fully-flushed code backend that, based on a UID provided to the program, will use a REST API to pull JSON, parse it, and update a database. For the awareness of the user, this also prints out status messages, object data, etc... Currently, the window I have looks like this:
![image](https://user-images.githubusercontent.com/35636043/128348423-63332603-c307-4c54-9acd-268fce731d66.png)

In WPF, I implemented a Console Manager class that launched a cmd prompt (windows specific) and redirected ```Console.WriteLine()```'s the output to that. However, I'm wondering what the method would be in order to get the console output to redirect into that textblock in the Avalonia UI? Any help or pointers would be much appreciated!

**Comment by maxkatz6** [comment, marked as answer]
> redirect into that textblock in the Avalonia UI

Can't answer on your question, but I don't see anything Avalonia UI related. 
On windows you can reuse same implementation, but now sure about other platforms. There is just no difference for avalonia where from it comes.

**Reply by sherlockson** [reply]
Gotcha, just wanted to make sure there wasn't a standard for the UI that was going to be able to be used cross-platform. Thanks for the help!
